addons/mail/controllers/attachment.py

```
@@ -65,50 +64,28 @@ def mail_attachment_upload(self, ufile, thread_id, thread_model, is_pending=Fals
-    def mail_attachment_delete(self, attachment_id, access_token=None, **kwargs):
+    def mail_attachment_delete(self, attachment_id, access_token=None):
         attachment = request.env["ir.attachment"].browse(int(attachment_id)).exists()
         if not attachment:
             request.env.user._bus_send("ir.attachment/delete", {"id": attachment_id})
             return
-        attachment_message = request.env["mail.message"].sudo().search(
-            [("attachment_ids", "in", attachment.ids)], limit=1)
-        message = ThreadController._get_message_with_access(attachment_message.id, mode="create", **kwargs)
-        if not request.env.user.share:
-            # Check through standard access rights/rules for internal users.
-            attachment._delete_and_notify(message)
-            return
-        # For non-internal users 2 cases are supported:
-        #   - Either the attachment is linked to a message: verify the request is made by the author of the message (portal user or guest).
-        #   - Either a valid access token is given: also verify the message is pending (because unfortunately in portal a token is also provided to guest for viewing others' attachments).
-        # sudo: ir.attachment: access is validated below with membership of message or access token
-        attachment_sudo = attachment.sudo()
-        if message:
-            if not ThreadController._can_delete_attachment(message, **kwargs):
-                raise NotFound()
-        else:
-            if (
-                not access_token
-                or not attachment_sudo.access_token
-                or not consteq(access_token, attachment_sudo.access_token)
-            ):
-                raise NotFound()
-            if attachment_sudo.res_model != "mail.compose.message" or attachment_sudo.res_id != 0:
-                raise NotFound()
-        attachment_sudo._delete_and_notify(message)
+        attachment._check_attachments_access([access_token])
```

The route originally raised `NotFound()` rather than whatever the check returned, maybe we should wrap in `replace_exceptions`.